signac/contrib/project.py

```diff
@@ -1171,7 +1181,7 @@ def check(self):
         logger.info("Checking workspace for corruption...")
         for job_id in self._find_job_ids():
             try:
-                sp = self.get_statepoint(job_id)
+                sp = self.open_job(id=job_id).sp
```

Comment:
I think this case is more complicated. If I remember correctly, `open_job(id)` cannot be run successfully if the job is corrupted, which defeats the purpose of this line. I am not sure what to do without further consideration.